File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head:
            return None
        cnt = 1
        temp = head
        while temp.next:
            cnt += 1
            temp = temp.next
        temp.next = head
        k = k % cnt
        end = cnt - k  # to get end of the list
        while end:
            temp = temp.next
            end -= 1
        # breaking last node link and pointing to NULL
        head = temp.next
        temp.next = None
        return head
       

File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head== null) return head;
        int cnt = 1;
        ListNode temp = head;
        while(temp.next != null){
            cnt += 1;
            temp = temp.next;
        }
        temp.next = head;
        k = k % cnt;
        int end = cnt - k;
        while(end != 0){
            temp = temp.next;
            end -= 1;
        }
        head = temp.next;
        temp.next = null;
        return head;
    }
}